src/wayland/handlers/keyboard_shortcuts_inhibit.rs

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+use crate::state::State;
+use smithay::{
+    delegate_keyboard_shortcuts_inhibit,
+    wayland::keyboard_shortcuts_inhibit::{
+        KeyboardShortcutsInhibitHandler, KeyboardShortcutsInhibitState, KeyboardShortcutsInhibitor,
+    },
+};
+
+impl KeyboardShortcutsInhibitHandler for State {
+    fn keyboard_shortcuts_inhibit_state(&mut self) -> &mut KeyboardShortcutsInhibitState {
+        &mut self.common.keyboard_shortcuts_inhibit_state
+    }
+
+    fn new_inhibitor(&mut self, inhibitor: KeyboardShortcutsInhibitor) {
+        // TODO: Restrict what apps can inhibit shortcuts
```

Comment:
https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/68 should allow hiding a protocol to specific apps.

But even then, I would argue it should be exposed by default for new applications, so we can show a permission dialog on first-use. We can decline any request anyway.

Reply:
For Flatpak, ideally "inhibit system shortcuts" could be listed as one of the permissions an app requires when you install it, just as permissions are shown on install on Android.